dbms/src/Flash/Mpp/MPPTask.h

```diff
@@ -123,12 +124,27 @@ class MPPTask : public std::enable_shared_from_this<MPPTask>
 
     void registerTunnels(const mpp::DispatchTaskRequest & task_request);
 
+    void initProcessListEntry(MPPTaskManagerPtr & task_manager);
+
     void initExchangeReceivers();
 
     String getErrString() const;
     void setErrString(const String & message);
 
+    MemoryTracker * getMemoryTracker() const;
+
 private:
+    struct ProcessListEntryHolder
+    {
+        std::shared_ptr<ProcessListEntry> process_list_entry;
```

Comment:
Might consider cache a raw pointer here to avoid get() overload? Just a suggestion.

Reply:
It is only used during compile time, so I think not very necessary to cache a raw pointer.